pkg/ccl/changefeedccl/sink.go

```diff
@@ -124,6 +124,21 @@ func getSink(
 	return sink, nil
 }
 
+func validateSinkOptions(opts map[string]string, sinkSpecificOpts map[string]struct{}) error {
```

Comment:
How would you feel about calling this function from the `getSink` call above so that all validation happens in 1 place?
You could add another helper in `getSink` which takes sink specific options, and a
call back responsible for actually creating the sink:

```
validateOptionsAndMakeSink := func(sinkOpts map[string]struct, makeSink func() (Sink, error)) (Sink, error) {
    if validateSinkOptions....; err != nil {
          
    }
    return makeSink()
}
```

Then, change newSink switch statement to delegate to that helper:
```
newSink .... {
    switch {
        ....
        case u.Scheme == changefeedbase.SinkSchemeKafka:
			return validateAndMakeSink(changefeedbase.KafkaOptions, func() (Sink, error) {
                return makeKafkaSink(ctx, sinkURL{URL: u}, feedCfg.Targets, feedCfg.Opts) 
            })
    }
}